addons/mail/static/src/models/composer_view.js

```diff
@@ -565,14 +569,15 @@ registerModel({
             if (this.messaging.currentPartner) { // not supported for guest
                 composer.thread.unregisterCurrentPartnerIsTyping({ immediateNotify: true });
             }
-            const postData = this._getMessageData();
-            const params = {
-                'post_data': postData,
-                'thread_id': composer.thread.id,
-                'thread_model': composer.thread.model,
-            };
             try {
                 composer.update({ isPostingMessage: true });
+                await composer.refreshLinkPreview(composer.textInputContent);
+                const postData = this._getMessageData();
+                const params = {
+                    'post_data': postData,
+                    'thread_id': composer.thread.id,
+                    'thread_model': composer.thread.model,
+                };
```

Comment:
Why this move? If RPC for preview fails I think we want to still post the message as normal, not cancel it.